component.go

```diff
@@ -116,8 +117,15 @@ type component struct {
 var _ Instance = &componentImpl{}
 
 // Main is interface implemented by an application's main component.
-// This component is created automatically by `weaver.Run`.
-type Main interface{}
+// This component is instantiated and its Main() method called by
+// `weaver.Run`.
+type Main interface {
+	// TODO(reviewer): Should we make this method optional?
```

Comment:
Making it optional would involve maintaining Main as an empty interface and using reflection to discover if the Main impl implements this method?

That seems less clear to me for the user. I like your alternatives below, however.

Reply:
Yes. The code is pretty simple, but the question is whether or not it is more complicated for the user. So I will try to just disable Ref[] and RPCs.